backend/hitas/views/condition_of_sale.py

```diff
@@ -0,0 +1,153 @@
+
+    class Meta:
+        model = ConditionOfSale
+        fields = [
+            "id",
+            "new_ownership",
+            "old_ownership",
+            "grace_period",
+            "fulfilled",
+        ]
+
+
+class ConditionOfSaleCreateSerializer(serializers.Serializer):
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        owner_uuid = lookup_id_to_uuid(self.context["view"].kwargs.get("owner_uuid"), Owner)
+
+        try:
+            attrs["owner"] = Owner.objects.prefetch_related(
+                Prefetch(
+                    "ownerships",
+                    Ownership.objects.select_related("apartment").only(
+                        "id",
+                        "percentage",
+                        "apartment__id",
+                        "apartment__first_purchase_date",
+                        "apartment__latest_purchase_date",
+                        "apartment__completion_date",
+                    ),
+                ),
+                Prefetch(
+                    "ownerships__apartment__sales",
+                    ApartmentSale.objects.only(
+                        "id",
+                        "purchase_date",
+                    ).order_by("purchase_date"),
+                ),
+            ).get(uuid=owner_uuid)
+        except Owner.DoesNotExist as error:
+            raise HitasModelNotFound(model=Owner) from error
+
+        return attrs
+
+    def create(self, validated_data: dict[str, Any]) -> list[ConditionOfSale]:
+        owner: Owner = validated_data.pop("owner")
+        ownerships = get_many_or_cached_prefetch(owner, "ownerships", Ownership)
+
+        to_save: list[ConditionOfSale] = []
+
+        for ownership in ownerships:
+            apartment = ownership.apartment
+
+            if apartment.is_new:
+                for other_ownership in ownerships:
+                    if ownership.id == other_ownership.id:
+                        continue
+
+                    to_save.append(ConditionOfSale(new_ownership=ownership, old_ownership=other_ownership))
+
+        if not to_save:
+            validated_data["conditions_of_sale"] = []
+            return validated_data["conditions_of_sale"]
+
+        # 'ignore_conflicts' so that we can create all missing conditions of sale if some already exist
+        ConditionOfSale.objects.bulk_create(to_save, ignore_conflicts=True)
+
+        # We have to fetch ownerships separately, since if only some conditions of sale in 'to_save' were created,
+        # the ids in the returned list from 'bulk_create' are not correct.
+        conditions_of_sale: list[ConditionOfSale] = list(
+            ConditionOfSale.objects.select_related("new_ownership__owner", "old_ownership__owner")
+            .filter(Q(new_ownership__owner=owner) | Q(old_ownership__owner=owner))
+            .all()
+        )
+
+        validated_data["conditions_of_sale"] = conditions_of_sale
+        return validated_data["conditions_of_sale"]
```

Comment:
Why set the value to the dict, then immediately return it 🤔

Reply:
No reason, I'll simplify this.